lib/Page.js

```diff
@@ -778,9 +778,9 @@ class Page extends EventEmitter {
     await this.$eval(selector, (element, values) => {
       if (element.nodeName.toLowerCase() !== 'select')
         throw new Error('Element is not a <select> element.');
-
       const options = Array.from(element.options);
-
+      if (!values.some(value => options.some(option => option.value === value)))
```

Comment:
I think this needs to use `values.every()` instead of `values.some()` to handle variadic inputs.

Maybe also an additional test would be helpful, since it's not so trivial to reason about.

Reply:
I suppose it would depend on the intended behaviour. I was more concerned about whether at least change is made to the select element. With values.every(), the behaviour becomes concerned about whether all intended changes are made. If we use values.every() we would be required, I think, to include a list of options that were not selected with the error message.

Reply:
I agree with having a special semantic for null would be a good way to include the old functionality of deselecting options. If I figure out a better way I will implement it.

Reply:
I'm of the opinion that, if we are doing these checks, then they should have the same semantics for one or several elements. Or better yet, follow the existing heuristic: if the `select` has the multiple attribute set, check all provided `values` exists within the options, otherwise only check the first.

If your goal is to only know how many options were changed/selected, I would prefer a different approach where we don't throw and instead just return an int for the total number of selections made.

Perhaps @aslushnikov can weight in on this, since it's quite subjective.

Reply:
I think the function can be considered to fail if it does not change the select element so would throw an error. That was why I thought throwing an error might be a good idea.

A return value is also a good solution. But if the function was to return a value, perhaps an array of all selected elements would be the best. For the deselected functionality, the function could then return an empty array. 

I find the heuristic of, on a select element without the multiple property set to true, assigning the element the first value in the provided array kind of counter intuitive. I could see two cases where a developer might provide multiple values to a select element that takes on only one value. 
1.) the developer believes the select element can actually take on multiple values. If you believe this case is how the function will be used, I think it should throw an error because the developers intended usage is not what will actually happen.
2.) the developer provides multiple values, some of which may not be options, and wants the select element to set the first value that is an option. 

What do you think? @aslushnikov @alixaxel